### PR TITLE
[SPARK-6226][MLLIB] add save/load in PySpark's KMeansModel

### PR DESCRIPTION
**Pull request by mengxr**
Use `_py2java` and `_java2py` to convert Python model to/from Java model. @yinxusen

### CODE REVIEW

**Comment by mengxr** [comment]
test this please

File: python/pyspark/mllib/common.py [hunk 1]
```diff
@@ -70,8 +70,8 @@ def _py2java(sc, obj):
         obj = _to_java_object_rdd(obj)
     elif isinstance(obj, SparkContext):
         obj = obj._jsc
-    elif isinstance(obj, list) and (obj or isinstance(obj[0], JavaObject)):
-        obj = ListConverter().convert(obj, sc._gateway._gateway_client)
+    elif isinstance(obj, list):
+        obj = ListConverter().convert([_py2java(sc, x) for x in obj], sc._gateway._gateway_client)
```
Comment (yinxusen) [inline python/pyspark/mllib/common.py]: @mengxr Let me guess, when I encounter an `Array[ndarray]`, this line of code will try to translate the `ndarray` one by one. For each `ndarray`, the `else` statement will be triggered, `PikleSerializer` will serialize the `ndarray`, then the helper function in JVM will turn it back to `Vector`.

**Comment by yinxusen** [comment]
@mengxr Don't we need extra unittest? Does doctest well enough?

File: python/pyspark/mllib/common.py (hunk 1 above)
Comment (mengxr) [inline python/pyspark/mllib/common.py]: Correct. Except that we pass in vectors because we call `_convert_to_vector` in `KMeansModel.save`.

**Comment by mengxr** [comment]
Not necessary. doctests are examples+unittests.

**Comment by mengxr** [comment]
Merged into master.

**Pull request closed without merging**